diff --git a/src/configuration/SimConfig.h b/src/configuration/SimConfig.h
index 95c29d7e..9741099c 100644
--- a/src/configuration/SimConfig.h
+++ b/src/configuration/SimConfig.h
@@ -155,7 +155,7 @@ namespace hemelb
         }
         unsigned int PropertyOutputCount() const
         {
-          return propertyOutputs.size();
+          return 0; //propertyOutputs.size();
         }
         extraction::PropertyOutputFile * GetPropertyOutput(unsigned int index) const
         {
