diff --git a/src/lb/lb.hpp b/src/lb/lb.hpp
index 0c096298..d7024f26 100755
--- a/src/lb/lb.hpp
+++ b/src/lb/lb.hpp
@@ -6919,7 +6919,7 @@ template<class LatticeType>
 #ifdef HEMELB_USE_GPU	// If exporting computation on GPUs
 				// Boolean variable for sending macroVariables to GPU global memory (avoids the if statement time%_Send_MacroVars_DtH==0 in the GPU kernels)
 				// Consider using: a) propertyCache.densityCache.RequiresRefresh() || propertyCache.velocityCache.RequiresRefresh() || propertyCache.wallShearStressMagnitudeCache.RequiresRefresh()
-				bool Write_GlobalMem = (mState->GetTimeStep()%frequency_WriteGlobalMem == 0) ? 1 : 0;
+				bool Write_GlobalMem = false;
 
 				// Get the HEMELB_KERNEL (ie LBGK, LBGKSL etc)
 				const std::string hemeKernel = QUOTE_CONTENTS(HEMELB_KERNEL);
