diff --git a/src/geometry/LatticeData.cc b/src/geometry/LatticeData.cc
index e4be5a38..fc5b70de 100755
--- a/src/geometry/LatticeData.cc
+++ b/src/geometry/LatticeData.cc
@@ -506,12 +506,9 @@ namespace hemelb
 			neighbourIndices.resize(latticeInfo.GetNumVectors() * localFluidSites);
 
 			// These will be private to each thread
-			std::vector< std::vector< std::vector< site_t > > > threadLocalSharedDistributionLocations(omp_get_max_threads());
-			#pragma omp parallel 
-			{
-				auto tid = omp_get_thread_num();
-				threadLocalSharedDistributionLocations[ tid ].resize(comms.Size());
-			}
+			const size_t MT = omp_get_max_threads();
+			std::vector< std::vector< std::vector< site_t > > > threadLocalSharedDistributionLocations(
+				MT, std::vector< std::vector< site_t > >(comms.Size()));
 #if 0
 			for (BlockTraverser blockTraverser(*this); blockTraverser.CurrentLocationValid(); blockTraverser.TraverseOne())
 			{
