diff --git a/src/elpa2/GPU/CUDA/interface_c_cuda_kernel.F90 b/src/elpa2/GPU/CUDA/interface_c_cuda_kernel.F90
index 1ed47ac8..661a993e 100644
--- a/src/elpa2/GPU/CUDA/interface_c_cuda_kernel.F90
+++ b/src/elpa2/GPU/CUDA/interface_c_cuda_kernel.F90
@@ -364,11 +364,9 @@ module cuda_c_kernel
       integer(kind=c_intptr_t) :: hh_tau ,hh
       integer(kind=c_intptr_t) :: my_stream
 
-!#ifdef WITH_NVIDIA_GPU_VERSION
-!      call launch_compute_hh_trafo_c_cuda_sm80_kernel_real_single(q, hh, hh_tau, nev, nb, ldq, ncols, my_stream)
-!#endif
-       print *,"the single precision kernel is not yet implemented"
-       stop 1
+#ifdef WITH_NVIDIA_GPU_VERSION
+      call launch_compute_hh_trafo_c_cuda_sm80_kernel_real_single(q, hh, hh_tau, nev, nb, ldq, ncols, my_stream)
+#endif
     end subroutine
 #endif /* WITH_NVIDIA_GPU_SM80_COMPUTE_CAPABILITY */
 
diff --git a/src/elpa2/compute_hh_trafo.F90 b/src/elpa2/compute_hh_trafo.F90
index 52b3aaf2..7736ab26 100644
--- a/src/elpa2/compute_hh_trafo.F90
+++ b/src/elpa2/compute_hh_trafo.F90
@@ -396,21 +396,21 @@ last_stripe_width, kernel, my_stream, success)
 #endif
 
 #if REALCASE == 1
-    !if (obj%gpu_setup%nvidiaComputeCapability >= 80) then
-    !  call launch_compute_hh_trafo_sm80_gpu_kernel_&
-    !       &MATH_DATATYPE&
-    !       &_&
-    !       &PRECISION&
-    !       &(a_dev + dev_offset, bcast_buffer_dev + dev_offset_1, &
-    !       hh_tau_dev + dev_offset_2, nl, nbw,stripe_width, ncols, my_stream)
-    !else
+    if (obj%gpu_setup%nvidiaComputeCapability >= 80) then
+      call launch_compute_hh_trafo_sm80_gpu_kernel_&
+           &MATH_DATATYPE&
+           &_&
+           &PRECISION&
+           &(a_dev + dev_offset, bcast_buffer_dev + dev_offset_1, &
+           hh_tau_dev + dev_offset_2, nl, nbw,stripe_width, ncols, my_stream)
+    else
       call launch_compute_hh_trafo_gpu_kernel_&
            &MATH_DATATYPE&
            &_&
            &PRECISION&
            &(a_dev + dev_offset, bcast_buffer_dev + dev_offset_1, &
            hh_tau_dev + dev_offset_2, nl, nbw,stripe_width, ncols, my_stream)
-    !end if
+    end if
 #endif
 
     endif

